src/en/news/blog/2024/silver-bullet-rocksdb-performance/index.md

```diff
@@ -0,0 +1,120 @@
+---
+title: "Silver bullet for RocksDB performance"
+date: 2024-02-13
+author: Stefan Kooman (hydro-b)
+tags:
+  - ceph
+  - osd
+  - rocksdb
+  - throughput
+  - iops
+  - performance
+---
+
+~~There is no~~ silver bullet regarding RocksDB performance. Now that I got your attention: you might be "lucky" when you are using upstream Ceph Ubuntu packages.
+
+### TL;DR
+Mark Nelson [found out](https://ceph.io/en/news/blog/2024/ceph-a-journey-to-1tibps/#fix-three) that [before](https://github.com/ceph/ceph/pull/54918) Pull request (PR) was merged, the build process did not properly propagate the `CMAKE_BUILD_TYPE` options to external projects built by Ceph - in this case, RocksDB. [Before](https://github.com/ceph/ceph/pull/24133), packages were built with the `RelWithDebInfo` to build a "performance" release package. While it has not been verified, it is possible that upstream Ceph Ubuntu packages have suffered from this since some version of Luminous.
+
+# Acknowledgments
+
+Thanks to Mark Nelson for finding and fixing this issue. Thanks to Kefu Chai for providing a fix for the build system. Thanks to Casey Bodley for taking care of creating the backport trackers. Thanks to my employer [BIT](https://www.bit.nl) for having me work on Ceph, and Els de Jong for editing.
+
+### So what does this mean?
+
+RocksDB performance is sub-optimal when build without `RelWithDebInfo`. This can be improved by running peformance packages. The actual performance increase depends on the cluster, but the RocksDB compaction is reduced by a factor of three. In some cases random 4K write performance is doubled. See these links [1](https://bugs.gentoo.org/733316) and [2](https://ceph.io/en/news/blog/2024/ceph-a-journey-to-1tibps/).
```

Comment:
suboptimal when built
This can be mitigated by installing "performance" package builds.